tests/test_inference_providers.py

```diff
@@ -925,3 +918,28 @@ def test_recursive_merge(dict1: Dict, dict2: Dict, expected: Dict):
     # does not mutate the inputs
     assert dict1 == initial_dict1
     assert dict2 == initial_dict2
+
+
+def test_get_provider_helper_auto(mocker):
+    """Test the 'auto' provider selection logic."""
+
+    mock_provider_a_helper = mocker.Mock(spec=TaskProviderHelper)
+    mock_provider_b_helper = mocker.Mock(spec=TaskProviderHelper)
+    original_providers = PROVIDERS.copy()
+    PROVIDERS["provider-a"] = {"test-task": mock_provider_a_helper}
+    PROVIDERS["provider-b"] = {"test-task": mock_provider_b_helper}
+
+    mocker.patch(
+        "huggingface_hub.inference._providers._fetch_inference_provider_mapping",
+        return_value={
+            "provider-a": mocker.Mock(),
+            "provider-b": mocker.Mock(),
+        },
+    )
+    helper = get_provider_helper(provider="auto", task="test-task", model="test-model")
+
+    # The helper should be the one from provider-a
+    assert helper is mock_provider_a_helper
+
+    PROVIDERS.clear()
+    PROVIDERS.update(original_providers)
```

Comment:
```suggestion
    PROVIDERS.pop("provider-a", None)
    PROVIDERS.pop("provider-b", None)
```

(nit) (just wary of concurrency issues if we clear + update since tests are done in parallel)